design/pipeline-triggers.md

```diff
@@ -0,0 +1,359 @@
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    trigger:
+      branches:
+      - master
+      - releases/*
+```
+
+You can specify branches and paths to include and exclude.
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    trigger:
+      branches:
+        include:
+        - features/*
+        exclude:
+        - features/experimental/*
+      paths:
+        exclude:
+        - README.md
+```
+
+
+If you have a lot of team members uploading changes often, then you might want to reduce the number of builds you're running. If you set batch to true, when a build is running, the system waits until the build is completed, then queues another build of all changes that have not yet been built.
+
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    trigger:
+      batch: true
+      branches:
+      - master
+```
+
+
+
+#### PR triggers
+Whenever a PR is raised on the repository, you can choose to trigger your pipeline using PR triggers. PR triggers are not enabled by defaut. You can enable PR triggers on the repository by defining `pr` trigger on the `repository` resource.
+
+
+#### Scenarios
+- I would like to trigger my pipeline only when a PR is targeted to `releases/*` branch of the repository.
+- I would like to trigger my pipeline only when a PR is targeted to the file path `Web/*` of the repository.
+
+
+#### Schema
+```yaml
+resources:         
+  repositories:
+  - repository: string    
+    source: string  
+    pr:        # Optional; pr triggers are disabled by default
+      branches:  # branch conditions to filter the events, optional; Defaults to all branches.
+        include: [ string ]  # branches to consider the trigger events, optional; Defaults to all branches.
+        exclude: [ string ]  # branches to discard the trigger events, optional; Defaults to none.  
+      paths:
+        include: [ string ]  # file paths to consider the trigger events, optional; Defaults to all branches.
+        exclude: [ string ]  # file paths to discard the trigger events, optional; Defaults to none.
+```
+
+#### Examples: PR triggers
+Unless you specify, `pr` triggers are disabled for your repository. You can enable pull request based pipeline runs.
+
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr: true
+```
+
+You can control the target branches for your pull request based pipeline runs by simple syntax.
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr: 
```

Comment:
where do pr fork settings live? (security problems in the file, which is why they live in the UI today)

Reply:
I feel this is also dependent on how triggers are considered across branches. 

If we go with today's behavior of triggers in Build YAML, since triggers are considered only from the default branch used in the pipeline, the security problems will not creep in. Only those with access to the default branch can change the trigger settings.

Reply:
As, discussed, we will take this incremental. Once we figure out the right place for pr fork policies for self-repos, we will replicate the same for other repositories.